### PR TITLE
gof5: init at 0.1.4

### PR DESCRIPTION
**Pull request by Leixb**
###### Description of changes

Open Source F5 BIG-IP VPN client for Linux, MacOS, FreeBSD and Windows.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 169422` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>gof5</li>
</ul>
</details>

File: pkgs/tools/networking/gof5/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  # code in order for it to work.
+  # See: https://github.com/kayrus/gof5/blob/674485bdf5a0eb2ab57879a32a2cb4bab8d5d44c/pkg/client/http.go#L172-L174
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Open Source F5 BIG-IP VPN client for Linux, MacOS, FreeBSD and Windows";
+    homepage = "https://github.com/kayrus/gof5";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ leixb ];
+    platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/gof5/default.nix:29]: buildGoModule already sets Linux + Darwin.

File: pkgs/tools/networking/gof5/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{ lib
+, fetchFromGitHub
+, buildGo117Module
+}:
+
+buildGo117Module rec {
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/gof5/default.nix:6]: Please remove the 117.

File: pkgs/tools/networking/gof5/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  vendorSha256 = null;
+
+  # The tests are broken and apparently you need to uncomment some lines in the
+  # code in order for it to work.
+  # See: https://github.com/kayrus/gof5/blob/674485bdf5a0eb2ab57879a32a2cb4bab8d5d44c/pkg/client/http.go#L172-L174
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Open Source F5 BIG-IP VPN client for Linux, MacOS, FreeBSD and Windows";
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/gof5/default.nix:25]:
```suggestion
    description = "Open Source F5 BIG-IP VPN client";
```

**Pull request merged**